primefaces-selenium/primefaces-selenium-components/src/main/java/org/primefaces/selenium/component/PickList.java

```diff
@@ -0,0 +1,481 @@
+    public void filterTargetList(final String filterValue) {
+        setFilterValue(targetFilter, filterValue);
+    }
+
+    /**
+     * Clears the source list filter
+     */
+    public void clearSourceListFilter() {
+        setFilterValue(sourceFilter, "");
+    }
+
+    /**
+     * Clears the target list filter
+     */
+    public void clearTargetListFilter() {
+        setFilterValue(targetFilter, "");
+    }
+
+    /**
+     * Filter using the Widget configuration "filterDelay" and "filterEvent" values.
+     *
+     * @param filterElement the source or target filter
+     * @param filterValue the value to set the filter
+     */
+    public void setFilterValue(WebElement filterElement, String filterValue) {
+        JSONObject cfg = getWidgetConfiguration();
+        String filterEvent = cfg.getString("filterEvent");
+        int filterDelay = cfg.getInt("filterDelay");
+        setFilterValue(filterElement, filterValue, filterEvent, filterDelay);
+    }
+
+    /**
+     * Filter the column using these values.
+     *
+     * @param filterElement the source or target filter
+     * @param filterValue the value to filter for
+     * @param filterEvent the event causing the filter to trigger such as "keyup" or "enter"
+     * @param filterDelay the delay in milliseconds if a "keyup" filter
+     */
+    public void setFilterValue(WebElement filterElement, String filterValue, String filterEvent, int filterDelay) {
+        PrimeSelenium.clearInput(filterElement, false);
+
+        Keys triggerKey = null;
+        filterEvent = filterEvent.toLowerCase(Locale.ROOT);
+        switch (filterEvent) {
+            case "keyup":
+            case "keydown":
+            case "keypress":
+            case "input":
+                if (filterDelay == 0) {
+                    filterElement = PrimeSelenium.guardAjax(filterElement);
+                }
+                break;
+            case "enter":
+                triggerKey = Keys.ENTER;
+                break;
+            case "change":
+            case "blur":
+                triggerKey = Keys.TAB;
+                break;
+            default:
+                break;
+        }
+
+        if (filterValue != null) {
+            ComponentUtils.sendKeys(filterElement, filterValue);
+        }
+        else {
+            // null filter press backspace to trigger the re-filtering
+            filterElement.sendKeys(Keys.BACK_SPACE);
+        }
+
+        if (triggerKey != null) {
+            filterElement.sendKeys(triggerKey);
+        }
+        else if (filterDelay > 0) {
+            PrimeSelenium.wait(filterDelay + 10);
+            PrimeSelenium.waitGui().until(PrimeExpectedConditions.animationNotActive());
+        }
+    }
+
+    /**
+     * Gets a list of the values in the source list
+     *
+     * @return the list of values
+     */
+    public List<String> getSourceListValues() {
+        return getSourceListElements().stream().filter(e -> PrimeSelenium.isElementDisplayed(e))
```

Comment:
This is doing double the work. The getSourceListElements() method also checks if the elements are displayed. Take out the isElementDisplayed check from the methods getSourceListValues(), getSourceListLabels(), getTargetListValues(), getTargetListLabels()